### PR TITLE
T130 in vision refactoring

### PR DESCRIPTION
**Pull request by jgroehl**

### CODE REVIEW

**Comment by jgroehl** [comment]
This pull request changes:

1. The detector positions of the InVision detection geometry such that they exactly match the positions in the iThera proprietary data files.
2. The efficiency with which optical simulations with multiple illuminations are run. The mcx binary output files are only written once.

File: simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+        if field_of_view_extent_mm is None:
+            self.field_of_view_extent_mm = np.asarray([-self.probe_width_mm/2,
+                                                       self.probe_width_mm/2,
+                                                       0, 0, 0, 100])
+        else:
+            self.field_of_view_extent_mm = field_of_view_extent_mm
+
+    def check_settings_prerequisites(self, global_settings) -> bool:
+        if global_settings[Tags.DIM_VOLUME_Z_MM] < (self.probe_height_mm + 1):
```
Comment (kdreher) [inline simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py:316]: Here we would add 1mm on both sides independently of the spacing. Better change this to one voxel

File: simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            self.logger.error("Volume z dimension is too small to encompass the device in simulation!"
+                              "Must be at least {} mm but was {} mm"
+                              .format((self.probe_height_mm + 1),
+                                      global_settings[Tags.DIM_VOLUME_Z_MM]))
+            return False
+        if global_settings[Tags.DIM_VOLUME_X_MM] < (self.probe_width_mm + 1):
```
Comment (kdreher) [inline simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py:322]: see above

File: simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        super(iTheraInvision256TFDetectionGeometry, self).__init__(
+             number_detector_elements=256,
+             detector_element_width_mm=0.635,
+             detector_element_length_mm=15,
+             center_frequency_hz=5e6,
+             bandwidth_percent=55,
+             sampling_frequency_mhz=40,
+             device_position_mm=device_position_mm)
+
+        self.positions = (np.asarray([[0.02890019, -0.02837304, 0.],
```
Comment (kdreher) [inline simpa/core/device_digital_twins/detection_geometries/ithera_invision_array.py:42]: Would it be possible to calculate these positions based on some basic parameters or do you specifically want these positions and we are allowed by the vendor to publish these positions?

**Review by kdreher — CHANGES_REQUESTED** [review]
Just two minor changes and one question, otherwise it looks good :)

**Pull request closed without merging**